src/librustc_target/abi/mod.rs

```diff
@@ -694,11 +694,7 @@ impl FieldPlacement {
 
     pub fn offset(&self, i: usize) -> Size {
         match *self {
-            FieldPlacement::Union(count) => {
-                assert!(i < count,
-                        "Tried to access field {} of union with {} fields", i, count);
```

Comment:
Would have been good to at least leave a FIXME in place here so that future readers of this code know there *should* be an assertion. I just had to rediscover this by browsing the rustc source starting somewhere in codegen.